llvm/lib/CodeGen/BranchFolding.cpp

```diff
@@ -90,10 +90,13 @@ namespace {
 
   /// BranchFolderPass - Wrap branch folder in a machine function pass.
 class BranchFolderLegacy : public MachineFunctionPass {
+  bool EnableTailMerge;
```

Comment:
Missing new pass manager handling (also should handle print/parse of the pass parameter). Also this should be done as a separate step

Reply:
Isn't the new pass manager part covered by https://github.com/llvm/llvm-project/pull/128858?